test/tools/unified-spec-runner/match.ts

```diff
@@ -486,6 +486,9 @@ function compareEvents(
         expect.fail(`expected ${path} to be instanceof CommandStartedEvent`);
       }
       compareCommandStartedEvents(actualEvent, expectedEvent.commandStartedEvent, entities, path);
+      if (expectedEvent.commandStartedEvent.hasServerConnectionId) {
+        expect(actualEvent).property('serverConnectionId').to.exist;
+      }
```

Comment:
I think we need an explicit "not exist" assertion when `hasServerConnectionId` is false, not just a true case, right?

Reply:
We do, in the case that hasServerConnectionId exists but is set to false. I just added it in.